### ISSUE TITLE
no-switch-case-fall-through should not require break if case exits with "return" statement

### ISSUE THREAD

**Issue by oocx** [palantir/tslint#1259]
### Bug Report

- __TSLint version__: `3.10.1`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: webpack tslint-loader

#### TypeScript code being linted

```ts
function test(x) {
    switch (x) {
        case "foo": {
            let someVar = 42;
            // ... more code ...
            return someVar;
        }
        case "bar":
            return 2;
    }
}
```

#### Actual behavior
src/bug.ts[8, 9]: expected a 'break' before 'case'

#### Expected behavior
No error, as there is no way for the "foo" case to fall through to the "bar" case.

It works correctly for case statements that do not use braces. However, as I need to create a new scope for variables I only need in the "case", I have to use braces.

**Comment by HolgerJeromin** [comment]
This is probably not because of the return, but the extra block: #785

**Comment by adidahiya** [comment]
yep, looks like a duplicate of #785
